### DISCUSSION TITLE
Click on a button

### DISCUSSION THREAD

**Discussion by KingofICQ** [gram-js/gramjs#524] [Q&A]
I need to detect that there is a buttons in an incoming message, and if there is one, click on first button.
Like that 
![4](https://user-images.githubusercontent.com/38585020/235153622-468a8e06-0b0b-44d1-b12f-28601ae5624e.png)

**Comment by rojvv** [comment]
Try this:

```ts
await event.message.buttons?.[0]?.[0].click({});
```

**Reply by Davithi** [reply]
const peer = event.originalUpdate.message.peerId;                                               
const msgId = event.originalUpdate.message.id;                                                  
const buttonData = showPhoneButton.data;                                                        
await client.invoke(new Api.messages.GetBotCallbackAnswer({                                     
       peer: peer,                                                                             
       msgId: msgId,                                                                            
       data: buttonData                      }));

**Comment by Davithi** [comment]
const peer = event.originalUpdate.message.peerId;                      
const msgId = event.originalUpdate.message.id;                      
const buttonData = showPhoneButton.data;                       
await client.invoke(new Api.messages.GetBotCallbackAnswer({                         
          peer: peer,                          
          msgId: msgId,                          
          data: buttonData                      
}));
